### DISCUSSION TITLE
Aligner does not open gap for deletions on some reads

### DISCUSSION THREAD

**Discussion by erhei01** [alexdobin/STAR#1605] [Q&A] [answered]
Hi,@alexdobin
I am using STAR 2.7.10a. thanks for your works!
I am interested the deletions in my samples, but it seems STAR do not open gaps for it even I use --alignEndsType EndToEnd  ,--scoreDelBase -1 and --scoreDelOpen 0.
Here I have 5 reads in abc.fq, and run STAR to align them to GRCh38 reference genome and gtf anontation. I wanted to get the CIGAR like "63M1D6M" but have got "unmapped" with "too many mismatches".

-------------------------here is the parameters-----------
STAR --runMode alignReads \
           --runThreadN 16 \
           --genomeDir ${INDEX} \
	   --sjdbGTFfile ${GTFFILE} \
           --alignEndsType EndToEnd \
           --outFilterMatchNmin 15 \
           --outFilterMismatchNmax 3 \
           --scoreDelBase -1 \
           --scoreDelOpen 0 \
           --readFilesIn ./abc.fq \
           --alignSJoverhangMin 16 \
           --outFilterMultimapNmax 10 \
           --outSAMtype BAM Unsorted \
           --outFileNamePrefix abc

------------------------here is the final.out----------------
                          Number of input reads |       5
                      Average input read length |       66
                                    UNIQUE READS:
                   Uniquely mapped reads number |       1
                        Uniquely mapped reads % |       20.00%
                          Average mapped length |       63.00
                       Number of splices: Total |       0
            Number of splices: Annotated (sjdb) |       0
                       Number of splices: GT/AG |       0
                       Number of splices: GC/AG |       0
                       Number of splices: AT/AC |       0
               Number of splices: Non-canonical |       0
                      Mismatch rate per base, % |       4.76%
                         Deletion rate per base |       0.00%
                        Deletion average length |       0.00
                        Insertion rate per base |       0.00%
                       Insertion average length |       0.00
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |       0
             % of reads mapped to multiple loci |       0.00%
        Number of reads mapped to too many loci |       0
             % of reads mapped to too many loci |       0.00%
                                  UNMAPPED READS:
  Number of reads unmapped: too many mismatches |       4
       % of reads unmapped: too many mismatches |       80.00%
            Number of reads unmapped: too short |       0
                 % of reads unmapped: too short |       0.00%
                Number of reads unmapped: other |       0
                     % of reads unmapped: other |       0.00%
                                  CHIMERIC READS:
                       Number of chimeric reads |       0
                            % of chimeric reads |       0.00%

-----------------------here is the bam file-----------------------
A00511:605:H32WMDSX3:4:2322:32072:35963	16	X	73824925	255	63M	*	0	0	TATTTTACAGTACCCAAAATAGCTGGTATGTAAGAGGCACTTAATATTCATTCATCCATCATT	FFFF,FFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFF	NH:i:1	HI:i:1	AS:i:56	nM:i:3	NM:i:3	MD:Z:59T0C0A1	jM:B:c,-1jI:B:i,-1

thanks for your help!

**Comment by erhei01** [comment]
I tried to increase the sensitivity by using "--seedSearchStartLmax 20 --winAnchorMultimapNmax 100 ", but no effect.

**Comment by erhei01** [comment]
I found this issue occurs when the deletions` distance to the end of reads is less than 12bp. 
For example,  with CIGAR "xM1DyM"  ,if x or y <= 12 the issue occurs, if both x and y > 12 STAR can discover the deletions correctly.
I tried many alignment parameters but did not find any parameters to solve this problem.

**Comment by alexdobin** [comment, marked as answer]
Hi @erhei01 

you found the right explanation... STAR cannot handle indels that have short "overhangs", i.e. are too close (<12-15b) to the read ends.

Cheers
Alex

**Reply by erhei01** [reply]
Hi,@alexdobin
Thanks for your help!
I am reading your paper _STAR: ultrafast universal RNA-seq aligner_ and _Mapping RNA-seq Reads with STAR_.
You said in _STAR: ultrafast universal RNA-seq aligner_ "In addition to detecting splice junctions, the MMP search,implemented in STAR, enables finding multiple **mismatches and indels,** as illustrated in Figure 1b." and "If the MMP search does not reach the end of a read because of the presence of one or more **mismatches**, the MMPs will serve as anchors in the genome that can be extended, allowing for alignments with mismatches."(Page 2 and Fig1.b)" "Although the sequential MMP search only finds the seeds **exactly matching** the genome, ..."(Page 3)
I do not think I understand whether indels are allowed in this extending procedure.
I guess this issue can be solved by allowing indels in this extending procedure?
May be this is the tradeoff between local sensitivity and run speed.
would you like to set options such as "--sensitive" or "--very-sensitive" to do this.

Cheers
Erhei

**Comment by alexdobin** [comment]
Hi @erhei01 

presently, the extension only works for mismatches. For indels, the seeds have to be found on both sides of the indel; hence the overhang cannot be too small. In principle, you can try to increase --seedMultimapNmax = 10000 by default, which can allow to reduce the min overhang size.

Cheers
Alex

**Reply by erhei01** [reply]
Hi,@alexdobin
Thank you very much!
Best wishes
Erhei
